src/app/containers/App/App.jsx

```diff
@@ -1,124 +1,66 @@
 import { renderRoutes } from 'react-router-config';
 import { withRouter } from 'react-router';
-import path from 'ramda/src/path';
-import getRouteProps from '#app/routes/utils/fetchPageData/utils/getRouteProps';
 import usePrevious from '#lib/utilities/usePrevious';
+import getRouteData from '#lib/utilities/getRouteData';
+import routes from '#app/routes';
 
-export const App = ({ routes, location, initialData, bbcOrigin, history }) => {
-  const {
-    service,
-    isAmp,
-    variant,
-    id,
-    assetUri,
-    errorCode,
-    route: { pageType },
-  } = getRouteProps(routes, location.pathname);
-
-  const { pageData, status, error, timeOnServer } = initialData;
+const updatePage = async ({ setState, pathname }) => {
+  const { data, isAmp, service, variant } = await getRouteData({
+    path: pathname,
+  });
 
-  const [state, setState] = useState({
-    pageData,
-    status,
+  setState({
+    ...data,
+    isAmp,
     service,
     variant,
-    id,
-    assetUri,
-    isAmp,
-    pageType,
-    error,
-    loading: false,
-    errorCode: errorCode || initialData.errorCode,
-    timeOnServer,
   });
+};
 
-  const isInitialMount = useRef(true);
-  const shouldSetFocus = useRef(false);
-
-  useEffect(() => {
-    if (shouldSetFocus.current) {
-      const contentEl = document.querySelector('h1#content');
-      if (contentEl) {
-        contentEl.focus();
-      }
-      shouldSetFocus.current = false;
-    }
-  }, [state.loading, state.pageData]);
+const setFocusOnMainHeading = () => {
+  const mainHeadingEl = document.querySelector('h1#content');
 
-  useEffect(() => {
-    if (isInitialMount.current) {
-      isInitialMount.current = false;
-    } else {
-      // Only update on subsequent page renders
-      const {
-        service: nextService,
-        variant: nextVariant,
-        id: nextId,
-        assetUri: nextAssetUri,
-        isAmp: nextIsAmp,
-        route,
-      } = getRouteProps(routes, location.pathname);
+  if (mainHeadingEl) {
+    mainHeadingEl.focus();
+  }
+};
 
-      let loaderTimeout;
-      const loaderPromise = new Promise(resolve => {
-        loaderTimeout = setTimeout(resolve, 500);
-      });
+export default withRouter(
+  ({ location, initialData, bbcOrigin, history, isAmp, service, variant }) => {
+    const { pathname } = location;
+    const isMounted = useRef(false);
+    const [state, setState] = useState({
+      ...initialData,
+      isAmp,
+      service,
+      variant,
+    });
 
-      loaderPromise.then(() => {
-        setState({
-          pageData: null,
-          status: null,
-          service: nextService,
-          variant: nextVariant,
-          id: nextId,
-          assetUri: nextAssetUri,
-          isAmp: nextIsAmp,
-          pageType: route.pageType,
-          loading: true,
-          error: null,
-          errorCode: null,
-          timeOnServer: null,
-        });
-      });
+    useEffect(() => {
```

Comment:
Does this remove the loading state?

Reply:
it does. was anything using it? can be added back in

Reply:
there was intention to have an actual loading component/ui, there was some work done a little while ago as well that suspends the page rendering for a small amount of time i.e. if the data fetch is super fast, we don't show a loading page. if the data fetch is slow we transition to a loading page.

However this really only matters when client side rendering.

Reply:
ah ok I'll add that stuff back in